flink-runtime/src/test/java/org/apache/flink/streaming/util/TwoInputStreamOperatorTestHarness.java

```diff
@@ -51,20 +56,20 @@ public TwoInputStreamOperatorTestHarness(
         super(operator, maxParallelism, numSubtasks, subtaskIndex);
 
         this.twoInputOperator = operator;
+        processor1 = RecordProcessorUtils.getRecordProcessor1(twoInputOperator);
+        processor2 = RecordProcessorUtils.getRecordProcessor2(twoInputOperator);
     }
 
     public void processElement1(StreamRecord<IN1> element) throws Exception {
-        twoInputOperator.setKeyContextElement1(element);
-        twoInputOperator.processElement1(element);
+        processor1.accept(element);
```

Comment:
It looks like `twoInputOperator.setKeyContextElement1(element);` is still needed, otherwise, the test will fail.

Reply:
I reverted the last commit, since every test is not properly consider & handle the threads relation.

We will have a new operator test harness for async state processing.